mne/epochs.py

```diff
@@ -2064,6 +2064,289 @@ def _drop_log_stats(drop_log, ignore=('IGNORED',)):
     return perc
 
 
+def make_metadata(events, event_id, tmin, tmax, sfreq,
+                  time_locked_events=None, keep_first=None, keep_last=None):
+    """Generate metadata from events for use with `~mne.Epochs`.
+
+    This function mimics the epoching process (constructs time windows around
+    events of interest) and collates information about other events that
+    occurred within those time windows. The information is returned as a
+    :class:`pandas.DataFrame` suitable for use as `~mne.Epochs` metadata: one
+    row per time-locked event, and columns indicating presence/absence and
+    latency of each ancillary event type.
+
+    Notably, the time window used for metadata generation need not correspond
+    to the time period used to create the `~mne.Epochs`; it may well be much
+    shorter or longer, if desired.
```

Comment:
I'd put this in `Notes` section, and maybe give there a very short example of why you'd want to do this.

Reply:
Addressed in caa36e9